File: TopCoder/src/main/java/com/leetcode/array/leetcode/ContainerWithMostWater.java
package com.leetcode.array.leetcode;

//https://leetcode.com/problems/container-with-most-water/
public class ContainerWithMostWater {

    public int maxArea(int[] height) {
        int low = 0;
        int high = height.length - 1;
        int max = Integer.MIN_VALUE;

        while (low < high) {
            max = Math.max(max, (high - low) * Math.min(height[low], height[high]));
            if (height[low] < height[high]) {
                low++;
            } else {
                high--;
            }
        }
        return max;
    }

    public static void main(String[] args) {
        int[] arr = {1,8,6,2,5,4,8,3,7};
        int result = new ContainerWithMostWater().maxArea(arr);
        System.out.println(result);
    }

}

File: TopCoder/python/arr/ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: 'List[int]') -> 'int':
        low = 0
        high = len(height)-1

        maxval = 0

        while low < high:
            maxval = max(maxval, (high-low)*min(height[low], height[high]))
            if height[low] < height[high]:
                low = low + 1
            else:
                high = high - 1

        return maxval

arr = [1,8,6,2,5,4,8,3,7]
sol = Solution()
res = sol.maxArea(arr)
print(res)